core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -180,6 +180,11 @@ class DAGScheduler(
   /** If enabled, FetchFailed will not cause stage retry, in order to surface the problem. */
   private val disallowStageRetryForTest = sc.getConf.getBoolean("spark.test.noStageRetry", false)
 
+  /** Number of consecutive fetch failures allowed before a stage is aborted */
+  private[scheduler] val maxConsecutiveFetchFailuresPerStage =
+    sc.getConf.getInt("spark.max.fetch.failures.per.stage",
```

Comment:
I agree with spark.stage.maxAttempts.  We could also do spark.stage.maxFailures but somehow that seems more easily confused with the number of task failures per stage.